d2render/d2mapengine/engine.go

```diff
@@ -86,52 +96,56 @@ func (v *Engine) GetRegionAt(x, y int) *EngineRegion {
 		if !region.Rect.IsInRect(x, y) {
 			continue
 		}
-		return &region
+		return region
 	}
 	return nil
 }
 
 func (v *Engine) Render(target *ebiten.Image) {
 	for _, region := range v.regions {
-		v.RenderRegion(region, target)
+		v.RenderRegion(*region, target)
 	}
 }
 
-func (v *Engine) RenderRegion(region EngineRegion, target *ebiten.Image) {
+func (v *Engine) GenTiles(region *EngineRegion) {
 	for y := 0; y < int(region.Region.TileHeight); y++ {
 		offX := -((y + region.Rect.Top) * 80) + (region.Rect.Left * 80)
 		offY := ((y + region.Rect.Top) * 40) + (region.Rect.Left * 40)
 		for x := 0; x < int(region.Region.TileWidth); x++ {
-			sx, sy := d2helper.IsoToScreen(x+region.Rect.Left, y+region.Rect.Top, int(v.OffsetX), int(v.OffsetY))
-			if sx > -160 && sy > -160 && sx <= 880 && sy <= 1000 {
-				v.RenderTile(region.Region, offX, offY, x, y, target)
-			}
+			region.Tiles = append(region.Tiles, &RegionTile{
+				tileX: x,
+				tileY: y,
+				offX:  offX,
+				offY:  offY,
+			})
+
 			offX += 80
 			offY += 40
 		}
 	}
 }
 
-func (v *Engine) RenderTile(region *Region, offX, offY, x, y int, target *ebiten.Image) {
-	tile := region.DS1.Tiles[y][x]
-	for i := range tile.Floors {
-		if tile.Floors[i].Hidden || tile.Floors[i].Prop1 == 0 {
-			continue
+func (v *Engine) RenderRegion(region EngineRegion, target *ebiten.Image) {
+	var tilesToRender []RegionTile
```

Comment:
A micro optimization here would be to:
```suggestion
	tilesToRender := make([]RegionTile, len(region.Tiles))
```

This will allow to allocate all the memory at once.  Append has the potential to create a lot of garbage otherwise.